ui/app/adapters/secret-v2-version.js

```diff
@@ -27,9 +28,9 @@ export default ApplicationAdapter.extend({
 
   findRecord() {
     return this._super(...arguments).catch(errorOrModel => {
-      // if it's a real 404, this will be an error, if not
-      // it will be the body of a deleted / destroyed version
-      if (errorOrModel instanceof DS.AdapterError) {
+      // if the response is a real 404 or if the secret is gated by a control group this will be an error,
+      // otherwise the response will be the body of a deleted / destroyed version
+      if (errorOrModel instanceof DS.AdapterError || errorOrModel instanceof ControlGroupError) {
```

Comment:
With `ControlGroup` now extending this, can we move the if statement back to how it was previously?

Reply:
aww yiss!